packages/rocketchat-ui-admin/client/SettingsCachedCollection.js

```diff
@@ -0,0 +1,31 @@
+import _ from 'underscore';
+
+export class PrivateSettingsCachedCollection extends RocketChat.CachedCollection {
+	constructor() {
+		super({
+			name: 'private-settings',
+			eventType: 'onLogged'
+		});
+	}
+
+	setupListener(eventType, eventName) {
+		super.setupListener(eventType, eventName);
+
+		// private settings also need to listen to a change of authorizationsfor the setting-based authorizations
+		RocketChat.Notifications[eventType || this.eventType](eventName || this.eventName, (t, record) => {
+			this.log('record received', t, record);
+			if (t === 'auth') {
+				if (! (RocketChat.authz.hasAllPermission([`change-setting-${ record._id }`, 'manage-selected-settings'])
```

Comment:
The concatenation `change-setting-${ record._id }` should be reusable function of the auth-package. However, as `rocketchat:lib` is lower than `rocketchat:auth`, this is not possible.
For simplicity (and worse maintainability) I just green-coded that...